portal-bridge/src/types/state/execution.rs

```diff
@@ -109,15 +182,56 @@ impl State {
         Ok(updated_accounts)
     }
 
+    fn execute_transaction(
+        &mut self,
+        tx: Transaction,
+        evm_evnironment: &Env,
+    ) -> anyhow::Result<BTreeSet<Address>> {
+        let block_number = u256_to_lower_u64(evm_evnironment.block.number);
+        let ResultAndState { state: changes, .. } = {
+            let mut evm = Evm::builder()
+                .with_ref_db(&self.database)
+                .with_env(Box::new(evm_evnironment.clone()))
+                .with_spec_id(get_spec_id(block_number))
+                .modify_tx_env(|tx_env| {
+                    tx_env.caller = tx
+                        .get_transaction_sender_address(
+                            block_number >= SPURIOUS_DRAGON_BLOCK_NUMBER,
```

Comment:
nit: I think that *designed* way to do this check is to do something like this:
```rust
get_spec_id(block_number).is_enabled_in(SpecId::SPURIOUS_DRAGON)
```

It might not matter in our case, unless in the future we want to support network other than mainnet (e.g. testnet or layer two).

If you decide to apply this change here, apply it in other places in this PR (I think there is one or two more).

Reply:
I think this out of scope of the PR